examples/next/pages/ui/hooks/useDataStore/filter/index.page.tsx

```diff
@@ -0,0 +1,56 @@
+import { Amplify, DataStore, Hub } from 'aws-amplify';
+import { Authenticator } from '@aws-amplify/ui-react';
+import * as React from 'react';
+import '@aws-amplify/ui-react/styles.css';
+
+import { AuthSignOutButton } from '../AuthActions';
+import awsExports from '../aws-exports';
+import CollectionAndRecord from './CollectionAndRecord';
+import { initializeTestData } from '../utils';
+
+Amplify.configure({
+  ...awsExports,
+  aws_appsync_graphqlEndpoint: 'https://fake-appsync-endpoint/graphql',
+});
+
+export default function App() {
+  const [isInitialized, setInitialized] = React.useState(false);
+  const initializeStarted = React.useRef(false);
+
+  React.useEffect(() => {
+    (window as any).LOG_LEVEL = 'DEBUG';
+    Hub.listen('ui', (data) => {
+      console.log(data);
+    });
+    Hub.listen('datastore', (data) => {
+      console.log(data);
+    });
+    const initializeTestUserData = async () => {
+      if (initializeStarted.current) {
+        return;
+      }
+      await DataStore.clear();
+      await initializeTestData();
+      setInitialized(true);
+    };
+
+    initializeTestUserData();
+    initializeStarted.current = true;
+  }, []);
+
+  if (!isInitialized) {
+    return null;
+  }
+
+  return (
+    <Authenticator>
```

Comment:
afaik you need to be authenticated to perform DataStore actions, the other DataStore tests follow same pattern but lmk, I don't have tons of context.

Reply:
I'm not sure myself but think it might be worth trying to test without the Authenticator/authentication steps to keep the datastore tests a bit more isolated. What do you think?

Reply:
Not sure I follow the concept of isolation in an integration test. Typical customer usecase involves authentication before performing CRUD operation on DataStore.